### ISSUE TITLE
Write a paragraph describing what each team member did

### ISSUE THREAD

**Issue by aszecsei** [aszecsei/ai-final-project#8]
@willem-dejong @Diego-Castaneda Comment below with a paragraph describing what you did on the project.

**Comment by willem-dejong** [comment]
Willem worked primarily on decision tree creation, data file read in, and model selection in ocaml. The data reader was made with the assumption of it being in CSV format, and was made so that all our main ocaml parts can use it (dtl, classifier, and our ocaml model selection). Willem made the dtl.ml which takes in various command line arguments. It reads in a data file for the training data, then creates a decision tree. He made it so that you can use dtl.ml for either normal dtl, or max-depth dtl. Willem also wrote up the large amount of hard coded information about the different sets of data (information about the attributes, their possible values, the possible classification, etc). He also created our ocaml version of model select using cross validation which takes in various optional and required command line arguments and prints out the selected tree, error rate in the training set, error rate in the validation set, and the chosen max depth.

**Comment by aszecsei** [comment]
Willem worked primarily on decision tree creation, data file read in, and model selection in ocaml. He created a data reader for CSV files, which was used by all of the OCaml binaries (\texttt{dtl}, \texttt{classifier}, and the OCaml-based model selection program). He helped program the \texttt{dtl} executable, which reads in a data file for the training data, then creates a decision tree. He also wrote up the large amount of hard coded information about the different sets of data, including information about the attributes, their possible values, the possible classification, etc. He also created the OCaml version of model selection using cross validation, which takes in various optional and required command line arguments and prints out the selected tree, error rate in the training set, error rate in the validation set, and the chosen max depth. He also elected to be the project recorder.

**Comment by Diego-Castaneda** [comment]
Diego's primary focus was on writing and testing the classifier specified in the project spec. In developing the \texttt{classify} binary executable, Diego had to use Ocaml's I/O functionality as well as methods generated from Yojson to read in the decision tree to be used for classification. Additionally he made use of the reader methods developed by Willem in order to get a list of examples to classify. As he developed the classifier he suggested that indexes referring to attributes be added as a field to the decision tree structure. Diego also wrote tests using Ocaml's testing framework, OUnit. For testing the classifier, he mocked up sample trees and examples for some of the selected datasets plus some made up scenario sets and verified that the classifier was indeed classifying examples correctly. Lastly, as the designated coordinator he setup daily meetings in the engineering building, and checked to make sure everyone knew their tasks and everyone else's.
